pkg/tokenreviews/tokenvalidation/validators/timeoutvalidator_test.go

```diff
@@ -171,27 +103,25 @@ func TestAuthenticateTokenTimeout(t *testing.T) {
 	// add some padding to all sleep invocations to make sure we are not failing on any boundary values
 	buffer := time.Nanosecond
 
-	tokenAuthenticator := NewTokenAuthenticator(accessTokenGetter, fakeUserClient.UserV1().Users(), NoopGroupMapper{}, nil, timeouts)
-
 	go timeouts.Run(stopCh)
 
 	// TIME: 0 seconds have passed here
 
 	// first time should succeed for all
-	checkToken(t, "testToken", tokenAuthenticator, accessTokenGetter, testClock, true)
+	checkToken(t, "testToken", timeouts, accessTokenGetter, testClock, false)
 	wait(t, putTokenSync)
 
-	checkToken(t, "quickToken", tokenAuthenticator, accessTokenGetter, testClock, true)
```

Comment:
why are these changing?  I'd rather try to limit changes in this first stage.

Reply:
The test was wrong, rather than checking the TimeoutValidator, it was checking whether the whole TokenAuthenticator worked.

I separated validators into their own package so that they form their own logical unit, and to underline that they are pluggable to the authenticator. If I kept the original behavior of the test, this would have been a cyclic import.

Instead, I added a bunch of new unit tests for the whole authenticator in the above package (see `TestTokenAuthenticator`)